receiver/jaegerreceiver/trace_receiver.go

```diff
@@ -269,44 +298,59 @@ func (jr *jReceiver) stopTraceReceptionLocked() error {
 	return err
 }
 
-const collectorReceiverTagValue = "jaeger-collector"
-const tchannelCollectorReceiverTagValue = "jaeger-tchannel-collector"
+func consumeTraceData(
+	ctx context.Context,
+	batches []*jaeger.Batch,
+	consumer consumer.TraceConsumer,
+) ([]*jaeger.BatchSubmitResponse, int, error) {
 
-func consumeTraceData(ctx context.Context, batches []*jaeger.Batch, consumer consumer.TraceConsumer) ([]*jaeger.BatchSubmitResponse, error) {
 	jbsr := make([]*jaeger.BatchSubmitResponse, 0, len(batches))
-
+	var consumerError error
+	numSpans := 0
 	for _, batch := range batches {
-		td, err := jaegertranslator.ThriftBatchToOCProto(batch)
-		// TODO: (@odeke-em) add this error for Jaeger observability
-		ok := false
-
-		if err == nil {
-			ok = true
-			td.SourceFormat = "jaeger"
-			consumer.ConsumeTraceData(ctx, td)
-			// We MUST unconditionally record metrics from this reception.
-			observability.RecordMetricsForTraceReceiver(ctx, len(batch.Spans), len(batch.Spans)-len(td.Spans))
+		numSpans += len(batch.Spans)
+		if consumerError != nil {
```

Comment:
I'm confused by this check on consumerError.  You're checking below and either appending ok or not ok response to jbsr.

I'm not 100% sure what the intention is here, but I think this if should be removed.

Reply:
Note that the old code was ignoring errors when calling `consumer.ConsumeTrace` so the client wouldn't be notified of data being not being accepted into the pipeline. 

The intention here is to correctly account for batches that were successfully submitted and batches that were not. The choice of stopping on the first failed batch is because most of the time such errors will be due to internal errors on the collector so it only puts more pressure on the collector to send more data. I will add a comment to try to  clarify the intent of short circuiting the loop.